File: daily-coding-challenge/2026/july/python/26_LetterDistance#350.py
"""

Letter Distance

Given two strings of equal length, return the sum of the shortest distances between each pair of characters.
  The input will only contain lowercase letters
  The alphabet is treated as a circle, so the distance between a and z is 1.

Tests:
  letter_distance("abc", "bcd") should return 3.
  letter_distance("abc", "xyz") should return 9.
  letter_distance("encrypt", "decrypt") should return 10.
  letter_distance("algorithm", "codeblock") should return 43.
  letter_distance("lobster", "penguin") should return 47.
  letter_distance("alligator", "crocodile") should return 55.

"""


def letter_distance(str1, str2):
    letters = "abcdefghijklmnopqrstuvwxyz"
    distance = 0

    for i in range(len(str1)):
        dist1 = abs(letters.find(str1[i]) - letters.find(str2[i]))
        dist2 = len(letters) - dist1
        distance += dist1 if dist1 < dist2 else dist2

    return distance


print(letter_distance("abc", "bcd"))  # 3.
print(letter_distance("abc", "xyz"))  # 9.
print(letter_distance("encrypt", "decrypt"))  # 10.
print(letter_distance("algorithm", "codeblock"))  # 43.
print(letter_distance("lobster", "penguin"))  # 47.
print(letter_distance("alligator", "crocodile"))  # 55.

File: daily-coding-challenge/2026/july/js/26_LetterDistance#350.js
/*

Letter Distance

Given two strings of equal length, return the sum of the shortest distances between each pair of characters.
  The input will only contain lowercase letters
  The alphabet is treated as a circle, so the distance between a and z is 1.

Tests:
  letterDistance("abc", "bcd") should return 3.
  letterDistance("abc", "xyz") should return 9.
  letterDistance("encrypt", "decrypt") should return 10.
  letterDistance("algorithm", "codeblock") should return 43.
  letterDistance("lobster", "penguin") should return 47.
  letterDistance("alligator", "crocodile") should return 55.

*/

function letterDistance(str1, str2) {
  let letters = "abcdefghijklmnopqrstuvwxyz";
  let distance = 0;

  for (let i = 0; i < str1.length; i++) {
    let dist1 = Math.abs(letters.indexOf(str1[i]) - letters.indexOf(str2[i]));
    let dist2 = letters.length - dist1;
    distance += dist1 < dist2 ? dist1 : dist2;
  }
  return distance;
}

console.log(letterDistance("abc", "bcd")); // 3.
console.log(letterDistance("abc", "xyz")); // 9.
console.log(letterDistance("encrypt", "decrypt")); // 10.
console.log(letterDistance("algorithm", "codeblock")); // 43.
console.log(letterDistance("lobster", "penguin")); // 47.
console.log(letterDistance("alligator", "crocodile")); // 55.
